File: app/src/main/java/com/example/vishistvarugeese/memory/DAO/NoteDAO.java
package com.example.vishistvarugeese.memory.DAO;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.example.vishistvarugeese.memory.Variables.Entity.Note;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface NoteDAO {

    @Query("SELECT * FROM Note")
    LiveData<List<Note>> getAllNotes();

    @Query("SELECT * FROM Note WHERE id=:note_id")
    Note getNote(int note_id);

    @Query("DELETE FROM Note")
    void deleteAll();

    @Insert
    void insert(Note note);

    @Query("DELETE FROM Note WHERE id=:note_id")
    void delete(int note_id);

    @Query("UPDATE Note SET title =:title, description =:description, timestamp =:timestamp WHERE id=:note_id")
    void update(String title, String description, String timestamp, int note_id);
}

File: app/src/main/java/com/example/vishistvarugeese/memory/notes/NoteDAO.kt
package com.example.vishistvarugeese.memory.notes

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface NoteDAO {
    @get:Query("SELECT * FROM Note")
    val allNotes: LiveData<List<Note>>

    @Query("SELECT * FROM Note WHERE id=:noteId")
    fun getNote(noteId: Int): Note?

    @Query("DELETE FROM Note")
    fun deleteAll()

    @Insert
    fun insert(note: Note?)

    @Query("DELETE FROM Note WHERE id=:noteId")
    fun delete(noteId: Int)

    @Query("UPDATE Note SET title =:title, description =:description, timestamp =:timestamp WHERE id=:noteId")
    fun update(title: String?, description: String?, timestamp: String?, noteId: Int)
}